### PR TITLE
test: End to end tests for cdc integration

### PR DESCRIPTION
**Pull request by pratikkumar90**
End to end tests for cdc integration feature.

### CODE REVIEW

File: projects/storefrontapp-e2e-cypress/cypress/helpers/vendor/cdc/cdc.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+export function login() {
+  cy.get('[id="gigya-login-form"]').within(() => {
+    cy.get('[placeholder="Email *"]').type(user.email);
+    cy.get('[placeholder="Password *"]').type(user.password);
+    cy.get('[class="gigya-input-submit"]').click();
+  });
+}
+
+export function verifyLoginSuccess() {
```
Comment (marlass) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/vendor/cdc/cdc.ts]: Duplicate of `verifyRegistrationSuccess`. We can remove one of these functions.

File: projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cdc/login/login.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import * as cdc from '../../../../helpers/vendor/cdc/cdc';
+import * as login from '../../../../helpers/login';
+
+describe('Login existing Customer', () => {
+  Cypress.on('uncaught:exception', (err, runnable) => {
```
Comment (marlass) [inline projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cdc/login/login.e2e-spec.ts]: What this error handler does?

File: projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cdc/profile-update/profile-update.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    return false;
+  });
+  before(() => {
+    cy.window().then((win) => win.sessionStorage.clear());
+    cy.visit('/cdc/login');
+  });
+
+  it('should update and redirect to login page', () => {
+    cdc.registerUser();
+    cdc.verifyRegistrationSuccess();
```
Comment (marlass) [inline projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cdc/profile-update/profile-update.e2e-spec.ts]: Can we move these 2 to `before` part?

**Review by marlass — CHANGES_REQUESTED** [review]
These tests are quite simple and to avoid some repetition and to improve performance.
I would group them into single test:

```
describe('CDC', () => {
	describe('Registration', () = > {
		it('should register new user', () = > {...})
	})
	describe('Login', () => {
		it('should be able to login with previously created account', () => {...})
	})
	describe('Profile details', () => {
		it('should be able to update it's own profile', () => {...})
	});
})
```

File: projects/storefrontapp-e2e-cypress/cypress/helpers/vendor/cdc/cdc.ts (hunk 1 above)
Comment (pratikkumar90) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/vendor/cdc/cdc.ts]: Create a generic method to verify login and registration behavior.

File: projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cdc/login/login.e2e-spec.ts (hunk 2 above)
Comment (pratikkumar90) [reply projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cdc/login/login.e2e-spec.ts]: I added this to ignore the webApplicationInject.js error that was coming up in the browser console, which was halting the processing of the next test steps.

File: projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cdc/profile-update/profile-update.e2e-spec.ts (hunk 3 above)
Comment (pratikkumar90) [reply projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cdc/profile-update/profile-update.e2e-spec.ts]: I have refactored it to re-use the same SampleUser details across the tests. First we register the customer, then use the same customer to login, and finally login and update the profile.

**Review by marlass — APPROVED** [review]

**Pull request merged**